src/modules/generators/fractals/hybridmulti.rs

```diff
@@ -0,0 +1,238 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use num_traits::Float;
+use math;
+use math::{Point2, Point3, Point4};
+use NoiseModule;
+use modules::Perlin;
+use super::util::*;
+
+/// Default noise seed for the BasicMulti noise module.
+pub const DEFAULT_HYBRIDMULTI_SEED: usize = 0;
+/// Default number of octaves for the BasicMulti noise module.
+pub const DEFAULT_HYBRIDMULTI_OCTAVES: usize = 6;
+/// Default frequency for the BasicMulti noise module.
+pub const DEFAULT_HYBRIDMULTI_FREQUENCY: f32 = 2.0;
+/// Default lacunarity for the BasicMulti noise module.
+pub const DEFAULT_HYBRIDMULTI_LACUNARITY: f32 = 2.0;
+/// Default persistence for the BasicMulti noise module.
+pub const DEFAULT_HYBRIDMULTI_PERSISTENCE: f32 = 0.25;
+/// Maximum number of octaves for the BasicMulti noise module.
+pub const HYBRIDMULTI_MAX_OCTAVES: usize = 32;
+
+/// Noise module that outputs hybrid Multifractal noise.
+///
+/// The result of this multifractal noise is that valleys in the noise should
+/// have smooth bottoms at all altitudes.
+#[derive(Clone, Debug)]
+pub struct HybridMulti<T> {
+    /// Seed.
+    pub seed: usize,
+
+    /// Total number of frequency octaves to generate the noise with.
+    ///
+    /// The number of octaves control the _amount of detail_ in the noise
+    /// function. Adding more octaves increases the detail, with the drawback
+    /// of increasing the calculation time.
+    pub octaves: usize,
+
+    /// The number of cycles per unit length that the noise function outputs.
+    pub frequency: T,
+
+    /// A multiplier that determines how quickly the frequency increases for
+    /// each successive octave in the noise function.
+    ///
+    /// The frequency of each successive octave is equal to the product of the
+    /// previous octave's frequency and the lacunarity value.
+    ///
+    /// A lacunarity of 2.0 results in the frequency doubling every octave. For
+    /// almost all cases, 2.0 is a good value to use.
+    pub lacunarity: T,
+
+    /// A multiplier that determines how quickly the amplitudes diminish for
+    /// each successive octave in the noise function.
+    ///
+    /// The amplitude of each successive octave is equal to the product of the
+    /// previous octave's amplitude and the persistence value. Increasing the
+    /// persistence produces "rougher" noise.
+    pub persistence: T,
+
+    sources: Vec<Perlin>,
+}
+
+impl<T: Float> HybridMulti<T> {
+    pub fn new() -> HybridMulti<T> {
+        HybridMulti {
+            seed: DEFAULT_HYBRIDMULTI_SEED,
+            octaves: DEFAULT_HYBRIDMULTI_OCTAVES,
+            frequency: math::cast(DEFAULT_HYBRIDMULTI_FREQUENCY),
+            lacunarity: math::cast(DEFAULT_HYBRIDMULTI_LACUNARITY),
+            persistence: math::cast(DEFAULT_HYBRIDMULTI_PERSISTENCE),
+            sources: build_sources(DEFAULT_HYBRIDMULTI_SEED, DEFAULT_HYBRIDMULTI_OCTAVES),
+        }
+    }
```

Comment:
Wondering if these `new` methods could be implementations of `Default`. Then we could do: `HybridMulti::default()`. Maybe not for this PR though...